third_party/golang/parse/lex.go

```diff
@@ -0,0 +1,262 @@
+package parse
```

Comment:
I borrowed this code from golang library, and modified it for my own use. Erictune says I should put this in third_party.